std/format.d

```diff
@@ -867,7 +867,7 @@ struct FormatSpec(Char)
                 // Get the matching balanced paren
                 for (uint innerParens;;)
                 {
-                    enforce(j < trailing.length,
+                    enforce(j + 1 < trailing.length,
```

Comment:
`enforceFmt` (= `enforce!FormatException`) is already used for indicating format string errors. I'll make use of this.